airflow/operators/python_operator.py

```diff
@@ -104,10 +97,21 @@ def execute(self, context):
                                  for k, v in airflow_context_vars.items()]))
         os.environ.update(airflow_context_vars)
 
-        if self.provide_context:
-            context.update(self.op_kwargs)
-            context['templates_dict'] = self.templates_dict
+        context.update(self.op_kwargs)
+        context['templates_dict'] = self.templates_dict
+
+        if {parameter for name, parameter
+           in signature(self.python_callable).parameters.items()
+           if str(parameter).startswith("**")}:
```

Comment:
Bit shorter: `{param for param in sig.parameters.values() if str(param).startswith("**")}`

or: `any(str(params).startswith("**") for params in sig.parameters.values())`

Reply:
Love it, thanks!